scripts/src/main/resources/scripts/functions

```diff
@@ -1824,53 +1830,142 @@ function doGetSoftwareVersion() {
+  else
+    if [ -n "${version}" ]
     then
-      software_version_variable="node"
+      if [ "${version}" = "latest" ]
+      then
+        doUpdateMirrors
+        version="$(head -n 1 "${DEVON_IDE_HOME}/mirrors/${software}/available-versions")"
+      fi
+      local software_version_variable="${software}"
+      # temporary workaround for #1098 should be fixed and removed with #1085
+      if [ "${software_version_variable}" = "nodejs" ]
+      then
+        software_version_variable="node"
+      fi
+      software_version_variable="$(echo "${software_version_variable}" | awk '{print toupper($0)}')_VERSION"
+      local devon_properties="${DEVON_IDE_HOME}/settings/devon.properties"
+      grep -v "${software_version_variable}" "${devon_properties}" > "${devon_properties}.tmp"
+      mv "${devon_properties}.tmp" "${devon_properties}"
+      echo "${software_version_variable}=${version}" >> "${devon_properties}"
+      doEchoInteraction "${software_version_variable}=${version} has been set in ${devon_properties}\nTo install that version call the following command:\ndevon ${commandlet} setup"
+    else
+      doError "You have to specify the version you want to set."
     fi
-    software_version_variable="$(echo "${software_version_variable}" | awk '{print toupper($0)}')_VERSION"
-    local devon_properties="${DEVON_IDE_HOME}/settings/devon.properties"
-    grep -v "${software_version_variable}" "${devon_properties}" > "${devon_properties}.tmp"
-    mv "${devon_properties}.tmp" "${devon_properties}"
-    echo "${software_version_variable}=${version}" >> "${devon_properties}"
-    doEchoInteraction "${software_version_variable}=${version} has been set in ${devon_properties}\nTo install that version call the following command:\ndevon ${commandlet} setup"
-  else
-    doError "You have to specify the version you want to set."
   fi
 }
 
 # $1: software
 function doListSoftwareVersions() {
   local software="${1}"
-  doUpdateMirrors
-  local versions_path="${DEVON_IDE_HOME}/mirrors/${software}/available-versions"
-  if [ ! -f "${versions_path}" ]
-  then
-    doError "No versions available at ${versions_path}"
-  elif [ "$(wc -l "${versions_path}" | awk '{print $1}')" -lt 20 ]
-  then
-    pr -4 -t -s "${versions_path}" | column -t
+  local edition
+  if [ -d "${DEVON_IDE_HOME}/urls" ]
+  then
+    doUpdateUrls
+    edition=$(doGetSoftwareEdition "${software}")
+    local versions_path="${DEVON_IDE_HOME}/urls/${software}/${edition}"
+    local anz_versions
+    anz_versions="$(find "${versions_path}" -mindepth 1 -maxdepth 1 -printf "%f\n" | wc -l)"
+    if [ ! -d "${versions_path}" ]
+    then
+      doError "No versions available at ${versions_path}"
+    elif [ "${anz_versions}" -lt 20 ]
+    then
+      find "${versions_path}" -mindepth 1 -maxdepth 1 -printf "%f\n" | sort -rV | pr -4 -t -s | column -t
+    else
+      find "${versions_path}" -mindepth 1 -maxdepth 1 -printf "%f\n" | sort -rV | head -20 | pr -4 -t -s | column -t
```

Comment:
Why do we call `find` twice wasting the results of the first call?
also quite some duplications and already before your change things like `pr -4 -t -s` and `column -t` had been duplicated. Now we have it 4 times.

Reply:
I didn't question the sense of the code, I just replaced "cat file" 1:1 with "find directories". Now, the code is optimized and looks easier.